influxdb/content.md

```diff
@@ -1,643 +1,262 @@
-
-For the container to run scripts, they must:
-
--	Be mounted in the container's `/docker-entrypoint-initdb.d` directory
--	Be named using the `.sh` file name extension
--	Be executable by the user running the `docker run` command--for example, to allow the current use to execute a script with `docker run`:
-
-	```bash
-	chmod +x ./scripts/<yourscript.sh>
-	```
-
-> #### Grant permissions to mounted files
->
-> By default, Docker runs containers using the user and group IDs of the user executing the `docker run` command. When files are bind-mounted into the container, Docker preserves the user and group ownership from the host system.
-
-The image exports a number of variables into the environment before executing scripts. The following variables are available for you to use in your scripts:
+### Start InfluxDB 3
 
--	`INFLUX_CONFIGS_PATH`: Path to the `influx` CLI connection configurations file written by `setup`/`upgrade`
--	`INFLUX_HOST`: URL to the `influxd` instance running `setup`/`upgrade`
--	`DOCKER_INFLUXDB_INIT_USER_ID`: ID of the initial admin user created by `setup`/`upgrade`
--	`DOCKER_INFLUXDB_INIT_ORG_ID`: ID of the initial organization created by `setup`/`upgrade`
--	`DOCKER_INFLUXDB_INIT_BUCKET_ID`: ID of the initial bucket created by `setup`/`upgrade`
+%%COMPOSE%%
 
-For example, to grant an InfluxDB 1.x client *write* permission to your initial bucket, create a `$PWD/scripts/setup-v1.sh` file that contains the following:
+The example `compose.yaml` defines services for InfluxDB 3 Core and InfluxDB 3 Enterprise, including file system volumes for data persistence. You can customize the example to suit your needs, such as changing the node ID or object store type (for example, `s3`).
 
-```bash
-#!/bin/bash
-set -e
-
-influx v1 dbrp create \
-  --bucket-id ${DOCKER_INFLUXDB_INIT_BUCKET_ID} \
-  --db ${V1_DB_NAME} \
-  --rp ${V1_RP_NAME} \
-  --default \
-  --org ${DOCKER_INFLUXDB_INIT_ORG}
-
-influx v1 auth create \
-  --username ${V1_AUTH_USERNAME} \
-  --password ${V1_AUTH_PASSWORD} \
-  --write-bucket ${DOCKER_INFLUXDB_INIT_BUCKET_ID} \
-  --org ${DOCKER_INFLUXDB_INIT_ORG}
-```
-
-Then, run the following command to start and set up InfluxDB using custom scripts:
+**InfluxDB 3 Core:**
 
 ```bash
-docker run -p 8086:8086 \
-     -v "$PWD/data:/var/lib/influxdb2" \
-     -v "$PWD/config:/etc/influxdb2" \
-     -v "$PWD/scripts:/docker-entrypoint-initdb.d" \
-     -e DOCKER_INFLUXDB_INIT_MODE=setup \
-     -e DOCKER_INFLUXDB_INIT_USERNAME=my-user \
-     -e DOCKER_INFLUXDB_INIT_PASSWORD=my-password \
-     -e DOCKER_INFLUXDB_INIT_ORG=my-org \
-     -e DOCKER_INFLUXDB_INIT_BUCKET=my-bucket \
-     -e V1_DB_NAME=v1-db \
-     -e V1_RP_NAME=v1-rp \
-     -e V1_AUTH_USERNAME=v1-user \
-     -e V1_AUTH_PASSWORD=v1-password \
-     %%IMAGE%%:2
+docker compose run influxdb3-core
```

Comment:
This is another good example of the kind of really basic command that IMO it's really reasonable to expect users to be able to figure out or intuit directly from the Compose YAML and the Compose documentation instead of providing explicitly (especially since most "real" deployments will probably want to run this via `compose up -d` or similar so it's not attached to an active terminal, so this doesn't even represent "real" usage very accurately).

Reply:
7f2c18e